airflow/models.py

```diff
@@ -119,12 +119,17 @@ def clear_task_instances(tis, session, activate_dag_runs=True):
             if ti.job_id:
                 ti.state = State.SHUTDOWN
                 job_ids.append(ti.job_id)
-        # todo: this creates an issue with the webui tests
-        # elif ti.state != State.REMOVED:
-        #     ti.state = State.NONE
-        #     session.merge(ti)
         else:
-            session.delete(ti)
+            try:
+                ti_retries = dag.get_task(ti.task_id).retries
+                ti.max_tries = ti.try_number + ti_retries
+            except (AttributeError, AirflowException):
```

Comment:
I still feel like these exceptions are too broad. Can we use if conditions instead? e.g. if dag == None?